streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -1363,6 +1365,16 @@ public void testWindowing() {
                 new KeyValueTimestamp<>(2, "L2+l2", 2002L),
                 new KeyValueTimestamp<>(3, "L3+l3", 2003L)
             );
+
+            //push two items with timestamp at grace edge; this should produce one join item, M0 is 'too late'
```

Comment:
We should add the "drop record sensor check == 0" before we intentionally push "late" data that should be dropped, and we should also test the sensor at the very end that is recorded the dropped records.